package/network/services/dnsmasq/files/dnsmasq.init

```diff
@@ -170,7 +170,26 @@ append_address() {
 }
 
 append_ipset() {
-	xappend "--ipset=$1"
+	if [ -x /sbin/fw4 ]; then
+		local domainlist=${1%/*}
+		local setlist=${1##*/}
+		local nftsets=""
+
+		IFS=,
+		for set in $setlist; do
+			local family=$(nft -t list set inet fw4 "$set" 2>&1 | sed -nre 's#^\t\ttype .*\bipv([46])_addr\b.*$#\1#p')
+
+			[ -n "$family" ] || \
+				logger -t dnsmasq "Cannot infer address family from not existing nftables set '$set'"
+
+			nftsets="${nftsets:+$nftsets,}${family:+$family#}inet#fw4#$set"
+		done
+		IFS=$' \t\n'
+
+		xappend "--nftset=$domainlist/$nftsets"
```

Comment:
Please add a check if `nftset` is supported by dnsmasq, similar to the `ipset` check. Otherwise dnsmasq should refuse to start.

Kevin has already such a check in hist dnsmasq bump [in his staging tree](https://git.openwrt.org/?p=openwrt/staging/ldir.git;a=shortlog;h=refs/heads/mine).